crates/protocol/protocol/src/batch/single.rs

```diff
@@ -580,4 +594,36 @@ mod tests {
             BatchValidity::Drop
         );
     }
+
+    #[test]
+    fn test_check_batch_drop_non_empty_interop_transition() {
+        // Gather a few test transactions for the batch.
+        let transactions = example_transactions();
+
+        // Construct a basic `SingleBatch`
+        let parent_hash = BlockHash::ZERO;
+        let epoch_num = 1;
+        let epoch_hash = BlockHash::ZERO;
+        let timestamp = 1;
+
+        let single_batch =
+            SingleBatch { parent_hash, epoch_num, epoch_hash, timestamp, transactions };
+
+        let cfg = RollupConfig {
+            max_sequencer_drift: 1,
+            block_time: 1,
+            hardforks: HardForkConfig { interop_time: Some(1), ..Default::default() },
+            ..Default::default()
+        };
+        let l1_blocks = vec![BlockInfo::default(), BlockInfo::default()];
+        let l2_safe_head = L2BlockInfo {
+            block_info: BlockInfo { timestamp: 0, ..Default::default() },
+            ..Default::default()
+        };
+        let inclusion_block = BlockInfo::default();
+        assert_eq!(
+            single_batch.check_batch(&cfg, &l1_blocks, l2_safe_head, &inclusion_block),
+            BatchValidity::Drop
+        );
```

Comment:
We also capture and assert logs in our Go tests to make sure we drop batches for the expected reason, not any other. Is there something like this available?

Reply:
Yeah; I can add that in. We have a trace collector.